### ISSUE TITLE
Python 3.7 breakpoint() usage

### ISSUE THREAD

**Issue by bnaul** [dask/dask#3693]
More of a discussion prompt than anything else: the [new](https://www.python.org/dev/peps/pep-0553/) `breakpoint()` functionality got me thinking about how it might be used to facilitate debugging dask workflows. For local dask code it's maybe easy enough already to switch to the synchronous scheduler and use pdb directly, but for distributed code especially it can be quite cumbersome. Maybe something in the vein of [launching IPython](http://distributed.readthedocs.io/en/latest/ipython.html#launch-ipython-within-dask-workers) could called on breakpoint and block until debugging is finished?

**Comment by mrocklin** [comment]
That would be pretty cool if someone can make it happen

**Comment by mrocklin** [comment]
In a side conversation @bnaul said the following:

> I tried web_pdb as suggested here https://hackernoon.com/python-3-7s-new-builtin-breakpoint-a-quick-tour-4f1aebc444c, it's actually quite nice but I feel like the ideal version would be: worker logs "run the following to command to connect to ipython: <...>" and then from an ipython shell I can copy/paste to get into that thread

**Comment by mrocklin** [comment]
Yeah, something like that might work.  I especially like that it doesn't require any work on the Dask side :)

I wonder if there is some remote PDB mechansim that could be enabled through IPython kernels.
